### PR TITLE
914 fix for process player / camera fix (desync)

### PR DESCRIPTION
**Pull request by Undid-Iridium**
As title states

### CODE REVIEW

**Comment by Undid-Iridium** [comment]
Let me test a little more variations, I will let you know when I am ready for merge.

File: Exiled.Events/Patches/Events/Scp914/UpgradingPlayer.cs [hunk 1]
```diff
@@ -107,12 +111,15 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
+            // Find InventoryUpgrade, and set position there.
             index = newInstructions.FindIndex(
                 instruction => instruction.LoadsField(Field(typeof(Scp914Upgrader), nameof(Scp914Upgrader.OnInventoryItemUpgraded))));
 
             Label continueLabel = generator.DefineLabel();
 
-            newInstructions[index + 13].labels.Add(continueLabel);
+            // Find iterator jump by going -3 from leave_s
+            int continueIndex = newInstructions.FindIndex(index, instruction => instruction.opcode == OpCodes.Leave_S) - 3;
+            newInstructions[continueIndex].labels.Add(continueLabel);
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Scp914/UpgradingPlayer.cs:122]:
```suggestion
            int continueIndex = newInstructions.FindIndex(index, instruction => instruction.opcode == OpCodes.Leave_S) - 3;

            newInstructions[continueIndex].labels.Add(continueLabel);
```

**Review by iopietro — CHANGES_REQUESTED** [review]
Also update the comments above `newInstructions.InsertRange(...)`.

File: Exiled.Events/Patches/Events/Scp079/ChangingCamera.cs [hunk 2]
```diff
@@ -60,42 +60,36 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
             for (int z = 0; z < newInstructions.Count; z++)
                 yield return newInstructions[z];
 
             ListPool<CodeInstruction>.Pool.Return(newInstructions);
         }
+
+        private static bool ChangingCameraEvent(Scp079CurrentCameraSync instance, ref float cost)
+        {
+            ChangingCameraEventArgs ev = new ChangingCameraEventArgs(Player.Get(instance.Owner), instance._switchTarget, cost);
+            Scp079.OnChangingCamera(ev);
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Scp079/ChangingCamera.cs:83]:
```suggestion
            ChangingCameraEventArgs ev = new ChangingCameraEventArgs(Player.Get(instance.Owner), instance._switchTarget, cost);

            Scp079.OnChangingCamera(ev);
```

**Review by iopietro — CHANGES_REQUESTED** [review]

**Review by louis1706 — CHANGES_REQUESTED** [review]
Testing show than 079 sometimes get stuck(unable to choose other camera). 

 (it's not always happening so not sure at 100% than it's from here)

**Review by iopietro — APPROVED** [review]

File: Exiled.Events/Patches/Events/Scp079/ChangingCamera.cs [hunk 3]
```diff
@@ -58,44 +58,35 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
         }
+
+        private static bool ChangingCameraEvent(Scp079CurrentCameraSync instance, ref float cost)
+        {
+            ChangingCameraEventArgs ev = new ChangingCameraEventArgs(Player.Get(instance.Owner), instance._switchTarget, cost);
+
+            Scp079.OnChangingCamera(ev);
+
+            instance.ServerSendRpc(true);
+            if(ev.IsAllowed)
```
Comment (IRacle1) [inline Exiled.Events/Patches/Events/Scp079/ChangingCamera.cs:87]:
```suggestion
            if (ev.IsAllowed)
```

**Review by IRacle1 — DISMISSED** [review]

File: Exiled.Events/Patches/Events/Scp079/ChangingCamera.cs [hunk 4]
```diff
@@ -58,44 +58,35 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
...
+        private static bool ChangingCameraEvent(Scp079CurrentCameraSync instance, ref float cost)
+        {
+            ChangingCameraEventArgs ev = new ChangingCameraEventArgs(Player.Get(instance.Owner), instance._switchTarget, cost);
+
+            Scp079.OnChangingCamera(ev);
+
+            instance.ServerSendRpc(true);
+            if (ev.IsAllowed)
+                cost = ev.AuxiliaryPowerCost;
+            return ev.IsAllowed;
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Scp079/ChangingCamera.cs:89]:
```suggestion
            instance.ServerSendRpc(true);

            if (ev.IsAllowed)
                cost = ev.AuxiliaryPowerCost;

            return ev.IsAllowed;
```

**Review by iopietro — CHANGES_REQUESTED** [review]

**Review by louis1706 — APPROVED** [review]

**Review by Thundermaker300 — APPROVED** [review]

**Pull request merged**